src/packet.rs

```diff
@@ -0,0 +1,49 @@
+//! Defines ITM packets and their possible contents.
+
+use heapless::Vec as HVec;
+use std::fmt::{self, Debug, Formatter};
+
+pub const MAX_PAYLOAD_SIZE: usize = 4;
+
+/// Represents a complete received packet.
+#[derive(Debug)]
+pub struct Packet {
+    /// The header byte received for this packet.
+    pub header: u8,
+
+    /// The kind (type) of this packet.
+    pub kind: Kind,
+}
+
+/// The type of a packet.
+#[derive(Debug)]
+pub enum Kind {
+    /// Data from a software application
+    Instrumentation(Instrumentation),
+
+    #[doc(hidden)]
+    /// External consumers shouldn't expect the public variants to
+    /// be complete: there are more variants to implement.
+    _NoExhaustiveMatch,
+}
+
+/// Contents of an Instrumentation packet, with data from a software application
+pub struct Instrumentation {
```

Comment:
Personally, I prefer to keep fields private and provide accessor methods to the fields (e.g. `fn payload(&self) -> &[u8]`). I go for public fields when I want to let the user create an instance of the struct. But I don't think we want to allow that in this case; also I think that as it is this struct lets the user construct invalid `Instrumentation` instances (port can only be in the range 0-31 iirc).

Reply:
Fixed.